pandas/core/strings/base.py

```diff
@@ -0,0 +1,225 @@
+import abc
+from typing import Pattern, Union
+
+import numpy as np
+
+from pandas._typing import Scalar
+
+
+class BaseStringArrayMethods(abc.ABC):
```

Comment:
Maybe necessary to add abstractmethod ``_str_map``.

Reply:
Is is necessary to keep both ``BaseStringArrayMethods`` and ``ObjectStringArrayMixin`` if the only class inherited from it is ``ObjectStringArrayMixin``. I  would suggest to move implementation from the child into the parent class and add abstract method ``_str_map``. Unless there is a clear vision on the future extension of ``BaseStringArrayMethods``.

Reply:
Our Arrow-backed StringArray (which is motivating this PR) will have separate implementations for most methods. It will inherit directly from BaseStringArrayMethods.